File: front-end/Norlan-Svelte-TypeScript-master/src/lib/models/IscrizioneCorso.ts
import type { Documento } from './Documento';
import type { StatoCorso } from '$lib/models/Enums';
/*
Entità relazionale per la gestione delle iscrizioni ai corsi.
Traccia le conferme di presenza, lo stato del corso e l'emissione dell'attestato documentale.
*/
export interface IscrizioneData {
	idUtente: number;
	idCorso: number;
	emailUtente: string;
	titoloCorso: string;
	dataOrarioCorso: string;
	presenzaConfermata: boolean;
	idDocumento?: number;
	documentoAttestato?: Documento;
	statoCorso?: StatoCorso;
	feedbackInviato?: boolean;
}

export class IscrizioneCorso {
	idUtente: number;
	idCorso: number;
	emailUtente: string;
	titoloCorso: string;
	dataOrarioCorso: string;
	presenzaConfermata: boolean;
	statoCorso?: string;
	idDocumento?: number;
	documentoAttestato?: Documento;
	feedbackInviato?: boolean;

	constructor(data: IscrizioneData) {
		this.idUtente = data.idUtente;
		this.idCorso = data.idCorso;
		this.emailUtente = data.emailUtente;
		this.titoloCorso = data.titoloCorso;
		this.dataOrarioCorso = data.dataOrarioCorso;
		this.presenzaConfermata = data.presenzaConfermata;
		this.statoCorso = data.statoCorso;
		this.feedbackInviato = data.feedbackInviato;

		if (data.idDocumento !== undefined) {
			this.idDocumento = data.idDocumento;
		}
		if (data.documentoAttestato !== undefined) {
			this.documentoAttestato = data.documentoAttestato;
		}
	}
}

File: back-end/norlan-client-portal-master/src/main/java/it/norlan/clientportal/model/IscrizioneCorso.java
package it.norlan.clientportal.model;

import jakarta.persistence.*;
import lombok.*;
import java.io.Serializable;

/**
 * Entità JPA che risolve la relazione Molti-a-Molti tra Utenti e Corsi (Join Table).
 * Utilizza una chiave primaria composita (@EmbeddedId) e traccia l'avanzamento
 * dell'utente (presenza confermata dal docente ed eventuale attestato generato).
 */

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Entity
@Table(name = "iscrizione_corso")
public class IscrizioneCorso {

    @EmbeddedId
    private IscrizioneId id;

    @ManyToOne(fetch = FetchType.LAZY)
    @MapsId("idUtente")
    @JoinColumn(name = "id_utente")
    private Utente utente;

    @ManyToOne(fetch = FetchType.LAZY)
    @MapsId("idCorso")
    @JoinColumn(name = "id_corso")
    private CorsoFormazione corso;

    @Column(name = "presenza_confermata")
    private Boolean presenzaConfermata = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_documento", referencedColumnName = "id_documento")
    private Documento documentoAttestato;

    @Embeddable
    @Data @NoArgsConstructor @AllArgsConstructor
    public static class IscrizioneId implements Serializable {
        private Integer idUtente;
        private Integer idCorso;
    }

    public void validaPresenza() {
        this.presenzaConfermata = true;
    }

    public void invalidaPresenza() {
        this.presenzaConfermata = false;
    }
}
